### ISSUE TITLE
Using RefererMode

### ISSUE THREAD

**Issue by bigcraig** [SimpleBrowserDotNet/SimpleBrowser#171]
In past example code Referer headers can be set explicitly. Now the Referer is get only. I am unable to generate Referer codes . I have tried using the unsafeurl Referer mode. Do you have any code examples??

**Comment by kevingy** [comment]
Hi @bigcraig,

> In past example code Referer headers can be set explicitly. Now the Referer is get only. I am unable to generate Referer codes .

What do you mean by "generate Referer codes"? What are you trying to do exactly?

> I have tried using the unsafeurl Referer mode. Do you have any code examples??

The best example is in the unit tests: 

SimpleBrowser.UnitTests.OnlineTests.RefererHeader
```
public void When_Testing_Referer_Unsafe_Url_Secure_Transition()
```

That will not, however, allow you to set the referer header. It only sets the mode that SimpleBrowser uses to determine the referer header for subsequent requests.

**Comment by bigcraig** [comment]
Thank you so much for such a quick response
I was able to generate Referer headers by changing the referer mode.The problem I had was I needed an intial http 200 return from the web server to generate the referer value forthe next navigation . I am automating some router configuration and I needed to provide a referer header for an older model. I needed to browse to the introduction page to get a http 200 response which would provide areferer value for the next navigation to configuration screen.  Previously I was going straight to the configuration screen and was getting a 504 error. The subsequent navigations also received a 504 error because the browser wasnt going to produce a referer header from a 504 response (Yes I have got better :)  )
It all became clear 1 minute after I emailed you
Sorry for the poor description of the issue.
SimpleBrowser is a wonderful piece of software.
Thank you so much for creating it 
(The one place I didnt look for a solution was the unit tests)

Craig Woollettmob: 0419 213 910twitter @woolncwSkype: BigCraig

Date: Tue, 17 May 2016 08:45:45 -0700
